### PR TITLE
TICK-370 Solves a plot stems problem with bokeh

### PR DESCRIPTION
**Pull request by stephanegaiffas**
Currently, we latest version of bokeh, many warnings are raised

### CODE REVIEW

File: tick/plot/plot_stem.py [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
+import warnings
 
 
 def stem(y: np.ndarray, show=True, title=None, x_range=None, y_range=None):
     """Stem plot using bokeh
 
     Parameters
     ----------
-    y : `np.ndarray`, shape=(n_coeffs,)
+    y : `numpy.ndarray`, shape=(n_coeffs,)
```
Comment (Mbompr) [inline tick/plot/plot_stem.py]: We have been writing `np.ndarray` everywhere, should we change it?

File: tick/plot/plot_stem.py [hunk 2]
```diff
@@ -31,18 +38,18 @@ def stem(y: np.ndarray, show=True, title=None, x_range=None, y_range=None):
...
     fig.segment(x, np.zeros(dim), x, y)
-    fig.title_text_font_size = "12pt"
+    fig.title.text_font_size = "12pt"
     if title is not None:
-        fig.title = title
+        fig.title.text = title
     if show:
         bk.show(fig)
-        return None
+        return fig
```
Comment (Mbompr) [inline tick/plot/plot_stem.py]: Should it return None of fig? With matpllotlib utilities we are returning fig everytime but it seems to be a bad idea. When you launch the function with show=True in a notebook it shows it twice...

File: tick/plot/plot_stem.py [hunk 3]
```diff
@@ -1,12 +1,13 @@
 import numpy as np
+import warnings
 
 
 def stem(y: np.ndarray, show=True, title=None, x_range=None, y_range=None):
```
Comment (Mbompr) [inline tick/plot/plot_stem.py]: should it be called stem_bokeh?

File: tick/plot/plot_stem.py [hunk 4]
```diff
@@ -85,3 +92,116 @@ def stems(ys: list, titles: list = None, show=True, sync_axes=True):
         return None
     else:
         return p
+
+
+def stems_matplotlib(ys: list, titles: list = None, show=True, sync_axes=True):
+    """Several stem plots with synchronized axes using bokeh rendering
```
Comment (Mbompr) [inline tick/plot/plot_stem.py]: matplotlib

File: tick/plot/plot_stem.py [hunk 5]
```diff
@@ -85,3 +92,116 @@ def stems(ys: list, titles: list = None, show=True, sync_axes=True):
...
+        if titles is not None:
+            title = titles[idx]
+        else:
+            title = None
+        fig = plt.stem(y, title=title, x_range=x_range, y_range=y_range)
+        figs.append(fig)
+        if idx == 0 and sync_axes:
+            x_range = fig.x_range
+            y_range = fig.y_range
+    p = bk.gridplot([[e] for e in figs])
```
Comment (Mbompr) [inline tick/plot/plot_stem.py]: is it using functions from bokeh? Why?

File: tick/plot/plot_stem.py [hunk 6]
```diff
@@ -85,3 +92,116 @@ def stems(ys: list, titles: list = None, show=True, sync_axes=True):
...
+    sync_axes : `bool`, default=`True`
+        If True, the axes of the stem plot are synchronized (available only
+        with ``rendering='bokeh'``
+
+    rendering : {'matplotlib', 'bokeh'}, default='matplotlib'
+        Rendering library. 'bokeh' might fail if the module is not installed.
+
+    Returns
+    -------
+    output : `bokeh.models.layouts.Column` or `None`
```
Comment (Mbompr) [inline tick/plot/plot_stem.py]: Output might be a matplotlib figure if rendering is set to matpllotlib isn't it?

File: tick/plot/plot_stem.py [hunk 7]
```diff
@@ -85,3 +92,116 @@ def stems(ys: list, titles: list = None, show=True, sync_axes=True):
...
+    p = bk.gridplot([[e] for e in figs])
+    if show:
+        bk.show(p)
+        return None
+    else:
+        return p
+
+
+def stems(ys: list, titles: list = None, show=True, sync_axes=True,
+          rendering: str='bokeh'):
```
Comment (Mbompr) [inline tick/plot/plot_stem.py]: Don't we use matpllotlib by default anywhere? Bokeh is not a requirement of tick

File: tick/plot/plot_stem.py [hunk 8]
```diff
@@ -85,3 +92,116 @@ def stems(ys: list, titles: list = None, show=True, sync_axes=True):
...
+        Rendering library. 'bokeh' might fail if the module is not installed.
+
+    Returns
+    -------
+    output : `bokeh.models.layouts.Column` or `None`
+        A bokeh object containing the layout of the plot, when ``show=False``,
+        None otherwise.
+    """
+
+    # TODO: put matplotlib rendering outsize. It has to return the figure if show=False
```
Comment (Mbompr) [inline tick/plot/plot_stem.py]: You don't use `stems_matplotlib`?

File: tick/plot/plot_stem.py (hunk 1 above)
Comment (stephanegaiffas) [inline tick/plot/plot_stem.py]: If think that using np.ndarray does not link to the numpy doc in sphinx's html while numpy.ndarray does, but it needs to be confirmed :)

File: tick/plot/plot_stem.py (hunk 2 above)
Comment (stephanegaiffas) [inline tick/plot/plot_stem.py]: This is why I return None here... (if I recall well)

File: tick/plot/plot_stem.py (hunk 4 above)
Comment (stephanegaiffas) [inline tick/plot/plot_stem.py]: I'll correct

File: tick/plot/plot_stem.py (hunk 5 above)
Comment (stephanegaiffas) [inline tick/plot/plot_stem.py]: ?

File: tick/plot/plot_stem.py (hunk 7 above)
Comment (stephanegaiffas) [inline tick/plot/plot_stem.py]: Yes, default should be matplotlib, I'll change that

File: tick/plot/plot_stem.py (hunk 6 above)
Comment (stephanegaiffas) [inline tick/plot/plot_stem.py]: Yes of course

File: tick/plot/plot_stem.py (hunk 8 above)
Comment (stephanegaiffas) [inline tick/plot/plot_stem.py]: Don't forget this is still WIP :)

File: tick/plot/plot_stem.py (hunk 2 above)
Comment (Mbompr) [inline tick/plot/plot_stem.py]: You say in the docstring you're doing so but in the code you are returning `fig`

File: tick/plot/plot_stem.py (hunk 5 above)
Comment (Mbompr) [inline tick/plot/plot_stem.py]: `p = bk.gridplot([[e] for e in figs])` in `stems_matplotlib` function

File: tick/plot/plot_stem.py (hunk 8 above)
Comment (Mbompr) [inline tick/plot/plot_stem.py]: Yes I've realized this afterward, but as you assigned me as a reviewer I thought it was ready

**Comment by stephanegaiffas** [comment]
Ready for a review

**Comment by stephanegaiffas** [comment]
Can someone reviews that ? (it's an easy and short one) @Mbompr @vinther @MaryanMorel

File: tick/plot/plot_stem.py [hunk 9]
```diff
@@ -1,89 +1,138 @@
...
     Returns
     -------
-    output : `bk.grid_plot`
-        A grid plot object
+    output : `bokeh.models.layouts.Column`, `matplotlib.pyplot.Figure` or `None`
+        Depending on the rendering type, returns a ``bokeh`` or a ``matplotlib``
+        object containing the layout of the plot, or `None` otherwise
     """
-    import bokeh.plotting as bk
+    from bokeh.plotting import show as bk_show
```
Comment (Mbompr) [inline tick/plot/plot_stem.py]: This should be imported in the bokeh part of the code. This idea is that tick should be able to run without bokeh installed

**Comment by Mbompr** [comment]
Is it used anywhere in the doc? It could be cool to ensure that the matplotlib version is always working.

**Pull request merged**